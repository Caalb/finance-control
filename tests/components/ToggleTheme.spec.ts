import {
  it,
  expect,
  describe,
  beforeEach,
} from 'vitest'

import { mount } from "@vue/test-utils";
import { mockNuxtImport } from '@nuxt/test-utils/runtime'

import { ToggleTheme } from "#components";

const createComponent = () => {
  return mount(ToggleTheme, {});
}

mockNuxtImport('useColorMode', () => {
  return () => ({ value: 'light'})
})

describe('ToggleTheme', () => {
  let wrapper: any;

  beforeEach(() => {
    wrapper = createComponent();
  });

  it('renders correctly', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
})
