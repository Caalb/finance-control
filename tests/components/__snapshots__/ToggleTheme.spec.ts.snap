// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ToggleTheme > renders correctly 1`] = `"<button class="relative inline-flex flex-shrink-0 border-2 border-transparent disabled:cursor-not-allowed disabled:opacity-50 focus:outline-none h-5 w-9 rounded-full focus-visible:ring-2 focus-visible:ring-yellow-500 dark:focus-visible:ring-yellow-400 focus-visible:ring-offset-2 focus-visible:ring-offset-white dark:focus-visible:ring-offset-gray-900 bg-yellow-500 dark:bg-yellow-400" data-testid="toggle-theme" id="headlessui-switch-mRW6KdlcYH_0" role="switch" type="button" tabindex="0" aria-checked="true" data-headlessui-state="checked"><span class="pointer-events-none relative inline-block rounded-full bg-white dark:bg-gray-900 shadow transform ring-0 transition ease-in-out duration-200 h-4 w-4 translate-x-4 rtl:-translate-x-4"><span class="opacity-100 ease-in duration-200 absolute inset-0 h-full w-full flex items-center justify-center transition-opacity" aria-hidden="true"><span class="i-heroicons-sun-20-solid h-3 w-3 text-yellow-500 dark:text-yellow-400"></span></span><span class="opacity-0 ease-out duration-100 absolute inset-0 h-full w-full flex items-center justify-center transition-opacity" aria-hidden="true"><span class="i-heroicons-mon-20-solid h-3 w-3 text-gray-400 dark:text-gray-500"></span></span></span></button>"`;
